mino/minogrpc/overlay.go

```diff
@@ -177,49 +206,45 @@ func (o overlayService) Stream(stream Overlay_StreamServer) error {
 					return
 				}
 				if err != nil {
-					err = xerrors.Errorf("failed to listen stream: %v", err)
+					err = xerrors.Errorf("failed to listen stream on child in "+
+						"overlay: %v", err)
 					fabric.Logger.Err(err).Send()
 					return
 				}
 			}
-		}()
-	}
-
-	// add the gateways based on the routing table
-	for k, v := range o.routingTable {
-		gateway, ok := sender.participants[v]
-		if !ok {
-			// TODO: handle this situation
-			fabric.Logger.Warn().Msg("fix static check until it's done")
-		}
-		sender.participants[k] = gateway
+		}(addr)
 	}
 
 	// listen on my own stream
 	go func() {
+
 		for {
 			err := listenStream(stream, &receiver, sender, o.addr)
 			if err == io.EOF {
+				<-ctx.Done()
```

Comment:
I'm not sure either, I added it during my deadlock chase.  If that's ok like that I'll remove that when I will be sure it still works against the DKG test in the other PR. Added in the TODO.

Reply:
Thing is I can't exist once the handler is done, because a node has two responsibilities: running the handler and relaying messages. So if I exist once the handler is done, then the node won't relay messages anymore, which can be problematic.